### PR TITLE
Build not just with Mpich but with full MPI matrix

### LINKED ISSUE #22: Return "make check" to build script
Travis-CI times out with parallel tests. Increasing the wall time will obviously address this. Sounds like the great leaders are developing plan to procure more capacity allowing the full suite of ESMF tests to run: https://github.com/conda-forge/conda-forge.github.io/issues/306.

`make check` should be returned to the build script once wall times are fungible.

### LINKED ISSUE #28: Build from tags not from git
Would it be possible to build from tags and not from git?

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #22 
closes #28

### CODE REVIEW

**Comment by xylar** [comment]
I didn't realize that `netcdf-fortran` is pinned to 4.4.x.  I need to rebuild that for the full MPI matrix: https://github.com/conda-forge/netcdf-fortran-feedstock/pull/43

**Comment by ocefpaf** [comment]
> I didn't realize that `netcdf-fortran` is pinned to 4.4.x. I need to rebuild that for the full MPI matrix: [conda-forge/netcdf-fortran-feedstock#43](https://github.com/conda-forge/netcdf-fortran-feedstock/pull/43)

We should probably update the pin.

**Comment by xylar** [comment]
> We should probably update the pin.

I was thinking the same but wasn't 100% sure what the side effects might be.  Do you want to set up a migrator?  I've not done anything like that before but could give it a go if you like.

**Comment by ocefpaf** [comment]
> Do you want to set up a migrator? I've not done anything like that before but could give it a go if you like.

You did it exactly how I would do it :smile:
Migrators are new to me as well.

**Comment by xylar** [comment]
@bekozi and @rokuingh, could you let me know if you have any concerns about this move to an MPI matrix (i.e. building with `openmpi` and without MPI via `mpiuni` in addition to `mpich`)?  I am finding that `esmf` doesn't play nice with other software I would like to have in my environment like `moab` because it isn't building with the right MPI stack and doesn't have the full MPI matrix.

**Comment by xylar** [comment]
Once this is done, I will move on to adding the MPI matrix to `esmpy`, though I have no experience using `esmpy` myself and might need some help there.

**Comment by xylar** [comment]
I noticed when I moved on to `esmpy` that is pulls in the source from a tarball that should also contain the correct source for ESMF.  I'm trying it out.  Fingers crossed...

**Comment by xylar** [comment]
After I switched to the tarball, the OSX openmpi build is now hanging.  Weird!

File: recipe/build.sh [hunk 1]
```diff
@@ -13,22 +13,26 @@ export ESMF_NETCDF="split"
 export ESMF_NETCDF_INCLUDE=${PREFIX}/include
 export ESMF_NETCDF_LIBPATH=${PREFIX}/lib
 
-export ESMF_COMM=mpich3
+export ESMF_CPP="${CC} -E -P -x c"
+export ESMF_F90=${FC}
+export ESMF_CXX=${CXX}
```
Comment (xylar) [inline recipe/build.sh]: With these environment variables set, it is no longer necessary to make links to `gcc` and `gfortran`.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,34 +1,62 @@
...
 
 package:
   name: esmf
   version: {{ version }}
 
 source:
-  git_url: git://git.code.sf.net/p/esmf/esmf
-  git_tag: ESMF_8_0_0
+  url: http://www.earthsystemmodeling.org/esmf_releases/public/ESMF_{{ ver }}/esmf_{{ ver }}_src.tar.gz
+  sha256: 75c34c41806e703551b6b79566edf34c23f1eebcf821749e5320e860e565d94f
```
Comment (xylar) [inline recipe/meta.yaml]: Following `esmpy`, I'm getting the source from a tarball instead of a git tag.

File: recipe/build.sh [hunk 3]
```diff
@@ -13,22 +13,36 @@ export ESMF_NETCDF="split"
...
+  export ESMF_CPP="${CPP} -E -P -x c"
+fi
+
+if [[ -z "$mpi" || "$mpi" == "nompi" ]]; then
+  export ESMF_F90=${FC}
+  export ESMF_CXX=${CXX}
+else
+  export ESMF_F90=mpif90
+  export ESMF_CXX=mpicxx
+fi
```
Comment (xylar) [inline recipe/build.sh]: If these 3 environment variables are defined, no links to `gcc` and `gfortran` should be needed.

**Comment by xylar** [comment]
Okay, back to building successfully!  Again, @bekozi and @rokuingh, let me know if you have any feedback.

**Comment by bekozi** [comment]
Looks fine to me. I think this can be merged. Thanks @xylar.

**Comment by bekozi** [comment]
@xylar I merged after I realized you don't have permissions (unitl now)! These are great improvements overall. Thanks again for modernizing the build.

**Comment by xylar** [comment]
Thanks @bekozi!

**Pull request merged**